gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,248 @@
+    def get_application_name(self, process: Process) -> Optional[str]:
+        pass
+
+
+_IP_PORT_RE = re.compile(r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(?:\:\d{2,5})?$")  # Matches against (ip(:port))
+_NON_AVAILABLE_ARG = str()
+
+
+def _get_cli_arg_by_name(args: List[str], arg_name: str, check_for_equals_arg: bool = False) -> str:
+    if arg_name in args:
+        return args[args.index(arg_name) + 1]
+
+    if check_for_equals_arg:
+        for arg in args:
+            arg_key, _, arg_val = arg.rpartition("=")
+            if arg_key == arg_name:
+                return arg_val
+
+    return _NON_AVAILABLE_ARG
+
+
+def _get_cli_arg_by_index(args: List[str], index: int) -> str:
+    try:
+        return args[index]
+    except IndexError:
+        return _NON_AVAILABLE_ARG
+
+
+def _append_python_module_to_proc_wd(process: Process, module_path: str) -> str:
+    if os.path.isabs(module_path):
+        return module_path
+
+    return os.path.join(process.cwd(), module_path)
+
+
+class _GunicornApplicationIdentifier(_ApplicationIdentifier):
+    def get_application_name(self, process: Process) -> Optional[str]:
+        # As of gunicorn documentation the WSGI module name most probably will come from the cmdline and not from the
+        # config file / environment variables (they added the option to specify `wsgi_app`
+        # in the config file only in version 20.1.0)
+
+        if "gunicorn" not in _get_cli_arg_by_index(process.cmdline(), 0) or "gunicorn" not in _get_cli_arg_by_index(
+            process.cmdline(), 1
+        ):
+            return None
+
+        for arg in process.cmdline():
+            # ":" might appear in ip:port or in wsgi app specification (module:func)
+            if ":" in arg:
+                if _IP_PORT_RE.match(arg):
+                    continue
+
+                return f"gunicorn-{_append_python_module_to_proc_wd(process, arg)}"
+
+        _logger.warning(
+            f"GunicornApplicationSeparator: matched against process {process} but couldn't find WSGI module"
+        )
+        return None
+
+
+class _UwsgiApplicationIdentifier(_ApplicationIdentifier):
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        if "uwsgi" not in _get_cli_arg_by_index(process.cmdline(), 0):
```

Comment:
[Same](https://github.com/Granulate/gprofiler/pull/242/files#r777154006)

Reply:
done for both